### DISCUSSION TITLE
Different behavior when inserting code

### DISCUSSION THREAD

**Discussion by devopg** [Milkdown/milkdown#663] [Q&A] [answered]
Hi. I insert code from IDE RIDER and milkdown recognizes perfectly, but then insert code from Visual Studio Code it displays incorrectly.

from Rider:
![image](https://user-images.githubusercontent.com/61819835/183834318-e8af7c1c-7e8f-495e-9b35-5c62fb8bad68.png)

fron VSC:
![image](https://user-images.githubusercontent.com/61819835/183834550-eca77fd4-7389-448b-8f49-4e80f9019e7f.png)

What does it depend on? How can this be fixed? 
Should I put the bug?

ps. example if use TinyMCE from all ide all work fine.

**Comment by Saul-Mirone** [comment, marked as answer]
VSCode's behavior is wired, it will put data into a key called 'vscode-editor-data' in clipboard, so you need to handle it by create a plugin.

**Reply by devopg** [reply]
Whose side of the plug-in?
Why does this work when pasted into other editors? (like tinycloud)
VSC is the top 1 , the most popular editor, maybe should make it work by default? Why discard he largest layer of users

**Reply by Saul-Mirone** [reply]
Maybe I'll consider support it next version.

**Reply by Saul-Mirone** [reply]
https://github.com/Saul-Mirone/milkdown/issues/665

**Reply by devopg** [reply]
cool. I think this will be a super useful update for most!!! )
